src/utils/dates.js

```diff
@@ -10,3 +10,22 @@ const ONE_DAY_IN_MILLISECONDS = 86400000;
 export function getFutureDate(offset) {
 	return new Date(Date.now() + offset * ONE_DAY_IN_MILLISECONDS);
 }
+
+function convertTimestamp(timestamp) {
+	if (timestamp) {
+		const { seconds, nanoseconds } = timestamp;
+
+		const timestampInMilliseconds = seconds * 1000 + nanoseconds / 1e9;
+
+		return timestampInMilliseconds;
+	}
+}
+
+export function subtractDates(todaysDate, dateLastPurchased) {
+	if (dateLastPurchased) {
+		const dateForCompare = convertTimestamp(dateLastPurchased);
```

Comment:
There's a f[irebase provided method](https://firebase.google.com/docs/reference/js/firestore_.timestamp#timestampfromdate) you can use instead of this custom function.

It would look something like

```javascript

import {
	Timestamp,
} from 'firebase/firestore';

const dateForCompare = Timestamp.fromDate(dateLastPurchased);

```

I mostly say this because a future task will involve some intricate sorting by date and leaning on a library can make things a bit more straightforward.

--------------------------------------------------------------

Another idea where the Timestamp class's built in methods come in handy could be when you save the `dateLastPurchased` do it using `Timestamp.now()` instead of the `new Date.now()` that you're currently using.

Then `todaysDate` would be a `Timestamp` instead of a JS `Date` and you could do the comparison line on line 28 as

```javascript

return todaysDate - dateForCompare < ONE_DAY_IN_MILLISECONDS;

```

without needing to do any date conversations.

Reply:
Great, totally makes sense, especially in the context of future tasks. Thank you for highlighting this method.

Reply:
@hun-ah I tried to implement the suggestions but ran into an error in the `subtractDates `function. I am going to hold off on my commit.

Reply:
Oh yeah I think my initial suggestion of at the end may have been ill advised.

```javascript

return todaysDate - dateForCompare < ONE_DAY_IN_MILLISECONDS;

```

I have one other idea that I can comment inline, but if you can get something else working with `Timestamp` that is good with me too.